src/ratelimit.ts

```diff
@@ -22,32 +22,32 @@ export const RateLimiter = (
     }
 
     if (
-      opt.store.has(ip) &&
-      timestamp - opt.store.get(ip)!.lastRequestTimestamp > opt.windowMs
+      await opt.store.has(ip) &&
+      timestamp - (await opt.store.get(ip)!).lastRequestTimestamp > opt.windowMs
     ) {
       opt.store.delete(ip);
     }
-    if (!opt.store.has(ip)) {
+    if (!await opt.store.has(ip)) {
```

Comment:
I would prefer you have any statements which require a callback (`await`) to be their own variable to avoid promise stacking and make for cleaner and understandable code.